core-bundle/src/EventListener/DataContainer/AdjustAutoforwardSubpaletteListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Contao\PageModel;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @Callback(table="tl_page", target="config.onload")
+ */
+class AdjustAutoforwardSubpaletteListener implements ServiceAnnotationInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function __invoke(DataContainer $dc): void
+    {
+        if ('tl_page' !== $dc->table || !$dc->id) {
+            return;
+        }
+
+        /** @var PageModel $pageAdapter */
+        $pageAdapter = $this->framework->getAdapter(PageModel::class);
+
+        if (null !== ($page = $pageAdapter->findByPk($dc->id)) && \in_array($page->type, ['error_401', 'error_403'], true)) {
```

Comment:
```php
if ((!$page = $pageAdapter->findByPk($dc->id)) || !\in_array($page->type, ['error_401', 'error_403'], true)) {
    return;
}
```

Early return FTW 😄

Reply:
But there is no code to return early from - so converting this to an "early return" would only cause more code lines, wouldn't it? :Þ

But sure, I will change it ;)

Reply:
It is not about lines of code. It is about complexity and how easy the code can be read and understood by humans.

```php
if ($a) {
    // …

    if ($b) {
        // …

        if ($c) {
            // do something
        }
    }
}
```

Compared to

```php
if (!$a) {
    return;
}

// …

if (!$b) {
    return;
}

// …

if (!$c) {
    return;
}

// do something
```

Which one do you find more understandable?

Reply:
Of course, I am all for early outs :). Just thought it would be overdoing it in this case. But as I said, I'll change it.